onadata/apps/api/viewsets/open_data_viewset.py

```diff
@@ -32,6 +33,36 @@ def replace_special_characters_with_underscores(data):
     return [re.sub(r"\W", r"_", a) for a in data]
 
 
+def clean_data_fields(data, xform):
+    """
+    Streamlines the row header fields
+    with the column header fields for the same form.
+    """
+    result = []
+    if data:
+        headers = xform.get_headers()
+        tableau_headers = remove_metadata_fields(headers)
+        for row in data:
+            # Include fields from the column headers
+            diff = set(tableau_headers).difference(set(row))
+            left_out_fields = {
+                field: None for field in diff}
+            row.update(left_out_fields)
+            result.append(row)
+    yield result
```

Comment:
Could we yield rows instead here ? _The generator as is will still cycle through the entire data object and return result as it was doing when it was a function.... I think it'll be better if we made this a generator that returns rows after cleaning them_

Reply:
cool.. I yielded the rows  and then used this genexp to determine the results